src/ORM/DataQuery.php

```diff
@@ -842,6 +842,14 @@ public function limit(?int $limit, int $offset = 0): static
         return $this;
     }
 
+    /**
+     * Get the current limit of this query.
+     */
+    public function getLimit(): array
+    {
+        return $this->query->getLimit();
+    }
```

Comment:
This new API is necessary for us to explicitly disallow calling `limit()` in an eagerloading manipulation, since at best allowing that would provide unexpected results.